src/app/ng2-auto-complete/ng2AutoComplete.module.ts

```diff
@@ -0,0 +1,28 @@
+import { NgModule } from '@angular/core';
+import { FormsModule } from "@angular/forms";
+import { XHRBackend, RequestOptions } from '@angular/http';
+import { BrowserModule  } from '@angular/platform-browser';
+
+import { AutoComplete } from './auto-complete';
+import { AutoCompleteComponent } from './auto-complete.component';
+import { AutoCompleteDirective } from './auto-complete.directive';
+
+import { ApiHttp } from '../auth/api-http.service';
+import { AuthService } from '../auth/auth.service';
+
+let ApiHttpProvider = {
+    provide: ApiHttp,
+    useFactory: (xhrBackend: XHRBackend, requestOptions: RequestOptions, authService: AuthService) => {
+        return new ApiHttp(xhrBackend, requestOptions, authService)
+    },
+    deps: [XHRBackend, RequestOptions, AuthService]
+};
+
+@NgModule({
+  imports: [ BrowserModule, FormsModule ],
+  entryComponents: [AutoCompleteComponent],
```

Comment:
An error after upgrading stumped me for a bit, where the new RC6 `ComponentFactoryResolver` couldn't find a component with the name of `AutocompleteComponent` even though it was properly defined and imported. Then I found some doc notes by browsing the RC6 changelog and related issues noting that any components dynamically created via the `ComponentFactoryResolver` need to be declared in this `entryComponents` property.